libs/sdk-backend-tiger/src/convertors/toBackend/afm/TotalsConverter.ts

```diff
@@ -41,6 +51,125 @@ export function convertTotals(def: IExecutionDefinition): Total[] {
+
+                    // Check for subtotals rows/columns
+                    if (
+                        attributeTotal.attributeIdentifier !== attributeIdentifiers[0] &&
+                        columnTotal.attributeIdentifier !== columnIdentifiers[0]
+                    ) {
+                        attributeDimensionIndex = attributeIdentifiers.findIndex(
+                            (aI) => aI === attributeTotal.attributeIdentifier,
+                        );
+                        columnDimensionIndex = columnIdentifiers.findIndex(
+                            (cI) => cI === columnTotal.attributeIdentifier,
+                        );
+
+                        hasRowAndColumnSubTotals = true;
+                    }
+                }
+
+                // extend marginal totals payload
+                if (hasRowAndColumnSubTotals) {
+                    const marginalTotal: Total[] = [
+                        {
+                            function: convertTotalType(attributeTotal.type),
+                            metric: attributeTotal.measureIdentifier,
+                            localIdentifier: marginalTotalLocalIdentifier(attributeTotal, columnIndex),
+                            totalDimensions: [
+                                {
+                                    dimensionIdentifier: "dim_0",
+                                    totalDimensionItems: attributeIdentifiers.slice(
+                                        0,
+                                        attributeDimensionIndex,
+                                    ),
+                                },
+                                {
+                                    dimensionIdentifier: "dim_1",
+                                    totalDimensionItems: columnIdentifiers
+                                        .slice(0, columnDimensionIndex)
+                                        .concat(MeasureGroupIdentifier),
+                                },
+                            ],
+                        },
+                    ];
+                    newTotals = [...newTotals, ...marginalTotal];
+                }
+            });
+
+            // extend grand totals payload
+            if (hasRowAndColumnGrandTotals) {
+                const grandTotal: Total[] = [
+                    {
+                        function: convertTotalType(attributeTotal.type),
+                        metric: attributeTotal.measureIdentifier,
+                        localIdentifier: grandTotalLocalIdentifier(attributeTotal, index),
+                        totalDimensions: [
+                            {
+                                dimensionIdentifier: "dim_1",
+                                totalDimensionItems: [MeasureGroupIdentifier],
+                            },
+                        ],
+                    },
+                ];
+                newTotals = [...newTotals, ...grandTotal];
+            }
+        });
+
+        return newTotals;
```

Comment:
maybe I'm missing something, but why do we need to return here? it should fall to the next return anyway.

Reply:
Good catch. Removed.